url/urlsearchparams-constructor.html

```diff
@@ -124,6 +127,30 @@
     params = new URLSearchParams('a%f0%9f%92%a9b=c');
     assert_equals(params.get('a\uD83D\uDCA9b'), 'c');
 }, 'Parse %f0%9f%92%a9');  // Unicode Character 'PILE OF POO' (U+1F4A9)
+
+;[
+  { "input": {"+": "%C2"}, "output": [[" ", "\uFFFD"]], "name": "object with +" },
+  { "input": {c: "x", a: "?"}, "output": [["c", "x"], ["a", "?"]], "name": "object with two keys" },
+  { "input": [["c", "x"], ["a", "?"]], "output": [["c", "x"], ["a", "?"]], "name": "array with two keys" }
+].forEach((val) => {
+    test(() => {
+        let params = new URLSearchParams(val.input),
+            i = 0
+        for (let param of params) {
+            assert_array_equals(param, val.output[i])
+            i++
+        }
+    }, "Construct with " + val.name)
+})
+
+test(() => {
+  let params1 = new URLSearchParams("a=b"),
+      params2 = new URLSearchParams()
+  params2[Symbol.iterator] = params1[Symbol.iterator]
+  let params3 = new URLSearchParams(params2)
+  assert_equals(params3.get("a"), params1.get("a"))
+  assert_equals(params3.get("a"), "b")
+}, "Evil constructor test")
```

Comment:
I don't think this works as expected. `params1[Symbol.iterator] === params2[Symbol.iterator] === URLSearchParams.prototype[Symbol.iterator]`. The assignment is a no-op. The iterator will look up the private data on the instance it's invoked on, so iterating over params2 will always give no params, so params3 should be empty.